plugin/path_config.go

```diff
@@ -100,11 +100,29 @@ func (b *backend) configFields() map[string]*framework.FieldSchema {
 }
 
 func (b *backend) configUpdateOperation(ctx context.Context, req *logical.Request, fieldData *framework.FieldData) (*logical.Response, error) {
+
+	conf, err := readConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	if conf == nil {
+		conf = new(configuration)
+		conf.ADConf = new(client.ADConf)
+	}
+
+	// Use the existing ldap client config if it is set
+	var existing *ldaputil.ConfigEntry
+	if conf.ADConf != nil && conf.ADConf.ConfigEntry != nil {
+		existing = conf.ADConf.ConfigEntry
+	}
```

Comment:
```suggestion
	// Use the existing ldap client config if it is set
	var existing *ldaputil.ConfigEntry
	if conf != nil && conf.ADConf != nil && conf.ADConf.ConfigEntry != nil {
		existing = conf.ADConf.ConfigEntry
	}
```

We shouldn't have to allocate memory for the pointer value. Otherwise the use of `new` above means that `conf.ADConf` will never be nil, even when it should.

Reply:
As Austin pointed out, the nil check on `conf.ADConf.ConfigEntry` would still gate this properly so we're good!